junit-jupiter-api/src/main/java/org/junit/jupiter/api/condition/MethodBasedCondition.java

```diff
@@ -63,11 +64,15 @@ private Method getConditionMethod(String fullyQualifiedMethodName, ExtensionCont
 		String[] methodParts = ReflectionUtils.parseFullyQualifiedMethodName(fullyQualifiedMethodName);
 		String className = methodParts[0];
 		String methodName = methodParts[1];
-		Class<?> clazz = ReflectionUtils.tryToLoadClass(className).getOrThrow(
+		Class<?> clazz = ReflectionUtils.tryToLoadClass(className, getClassLoader(context)).getOrThrow(
```

Comment:
```suggestion
		Class<?> clazz = ReflectionUtils.tryToLoadClass(className, testClass.getClassLoader()).getOrThrow(
```

The test class is required to be present.

In fact, it's already accessed in the above if-block before this line is invoked.

So, please assign the test class to a local variable, use it within the if-block, and use it again here.